contracts/metadata/metadata.cairo

```diff
@@ -0,0 +1,647 @@
+        alloc_locals;
+        if (index == resources_len) {
+            return (values_index,);
+        }
+        let resource = resources[index];
+        if (resource == 0) {
+            return (values_index,);
+        }
+        if (resource == 1) {
+            assert values[values_index + 3] = Utils.ResourceNames.Wood;
+        }
+        if (resource == 2) {
+            assert values[values_index + 3] = Utils.ResourceNames.Stone;
+        }
+        if (resource == 3) {
+            assert values[values_index + 3] = Utils.ResourceNames.Coal;
+        }
+        if (resource == 4) {
+            assert values[values_index + 3] = Utils.ResourceNames.Copper;
+        }
+        if (resource == 5) {
+            assert values[values_index + 3] = Utils.ResourceNames.Obsidian;
+        }
+        if (resource == 6) {
+            assert values[values_index + 3] = Utils.ResourceNames.Silver;
+        }
+        if (resource == 7) {
+            assert values[values_index + 3] = Utils.ResourceNames.Ironwood;
+        }
+        if (resource == 8) {
+            assert values[values_index + 3] = Utils.ResourceNames.ColdIron;
+        }
+        if (resource == 9) {
+            assert values[values_index + 3] = Utils.ResourceNames.Gold;
+        }
+        if (resource == 10) {
+            assert values[values_index + 3] = Utils.ResourceNames.Hartwood;
+        }
+        if (resource == 11) {
+            assert values[values_index + 3] = Utils.ResourceNames.Diamonds;
+        }
+        if (resource == 12) {
+            assert values[values_index + 3] = Utils.ResourceNames.Sapphire;
+        }
+        if (resource == 13) {
+            assert values[values_index + 3] = Utils.ResourceNames.Ruby;
+        }
+        if (resource == 14) {
+            assert values[values_index + 3] = Utils.ResourceNames.DeepCrystal;
+        }
+        if (resource == 15) {
+            assert values[values_index + 3] = Utils.ResourceNames.Ignium;
+        }
+        if (resource == 16) {
+            assert values[values_index + 3] = Utils.ResourceNames.EtherealSilica;
+        }
+        if (resource == 17) {
+            assert values[values_index + 3] = Utils.ResourceNames.TrueIce;
+        }
+        if (resource == 18) {
+            assert values[values_index + 3] = Utils.ResourceNames.TwilightQuartz;
+        }
+        if (resource == 19) {
+            assert values[values_index + 3] = Utils.ResourceNames.AlchemicalSilver;
+        }
+        if (resource == 20) {
+            assert values[values_index + 3] = Utils.ResourceNames.Adamantine;
+        }
+        if (resource == 21) {
+            assert values[values_index + 3] = Utils.ResourceNames.Mithral;
+        }
+        if (resource == 22) {
+            assert values[values_index + 3] = Utils.ResourceNames.Dragonhide;
+        }
+
+        let trait_key = 639351341392214529084052888304630306;  // {"trait_type":"
+        let resource_key = 389105490742926503125548;  // Resource",
```

Comment:
I think these two can be merged into one. `len('{"trait_type":"Resource",') <= 31`. Similar in the next two functions.